contrib/create-waku/src/renderTemplate.ts

```diff
@@ -0,0 +1,34 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { deepMerge } from './deepMerge'
+
+export function renderTemplate(src: string, dest: string) {
+  const stats = fs.statSync(src);
+
+  if (stats.isDirectory()) {
+    fs.mkdirSync(dest, { recursive: true });
+    for (const file of fs.readdirSync(src)) {
+      renderTemplate(path.resolve(src, file), path.resolve(dest, file));
+    }
+    return;
+  }
+
+  const filename = path.basename(src);
+
+  if (filename === 'package.json' && fs.existsSync(dest)) {
+    // merge instead of overwriting
+    const pkg = deepMerge(
```

Comment:
why do we need deep merge here?

Reply:
For using ``projectName`` as ``name`` key in ``package.json`` we created the ``package.json`` already, so here we have to deep merge rest of the fields.

Reply:
Can't it be shallow merge??

Reply:
Yes, its working.